renderer/reducers/account.js

```diff
@@ -0,0 +1,266 @@
+    const isAccountPasswordEnabled = accountSelectors.isAccountPasswordEnabled(getState())
+    if (!isAccountPasswordEnabled) {
+      dispatch({ type: LOGIN_SUCCESS })
+    }
+    dispatch({ type: INIT_ACCOUNT_SUCCESS })
+  } catch (error) {
+    dispatch({ type: INIT_ACCOUNT_FAILURE, error: error.message })
+  }
+}
+
+/**
+ * setPassword - Updates wallet password.
+ *
+ * @param {string} password new password
+ * @returns {Function} Thunk
+ */
+const setPassword = password => async dispatch => {
+  dispatch(send('setPassword', { password: await sha256digest(password) }))
+}
+
+/**
+ * changePassword - Changes existing password.
+ *
+ * @param {object} params password params
+ * @param {string} params.newPassword new password
+ * @param {string} params.oldPassword old password
+ * @returns {Function} Thunk
+ */
+export const changePassword = ({ newPassword, oldPassword }) => async dispatch => {
+  try {
+    const intl = getIntl()
+    await dispatch(requirePassword(oldPassword))
+    await dispatch(setPassword(newPassword))
+    dispatch(closeDialog(CHANGE_PASSWORD_DIALOG_ID))
+    dispatch(showNotification(intl.formatMessage(messages.account_password_updated)))
+  } catch (error) {
+    dispatch({ type: LOGIN_FAILURE, error: error.message })
+  }
+}
+
+/**
+ * enablePassword - Enables app-wide password protection.
+ *
+ * @param {string} password to be used further on.
+ * @returns {Function} Thunk
+ */
+export const enablePassword = ({ password }) => async dispatch => {
+  try {
+    const intl = getIntl()
+    dispatch(setPassword(password))
+    dispatch(
+      saveConfigOverrides({
+        password: {
+          active: true,
+        },
+      })
+    )
+    dispatch(closeDialog(PASSWORD_SET_DIALOG_ID))
+    dispatch(showNotification(intl.formatMessage(messages.account_password_enabled)))
+  } catch (error) {
+    dispatch({ type: LOGIN_FAILURE, error: error.message })
+  }
+}
+
+/**
+ * enablePassword - Disables app-wide password protection.
+ *
+ * @param {string} password current password.
+ * @returns {Function} Thunk
+ */
+export const disablePassword = ({ password }) => async dispatch => {
+  try {
+    const intl = getIntl()
+    await dispatch(requirePassword(password))
+    dispatch(send('deletePassword'))
+    dispatch(
+      saveConfigOverrides({
+        password: {
+          active: false,
+        },
+      })
+    )
+    dispatch(closeDialog(PASSWORD_PROMPT_DIALOG_ID))
+    dispatch(showNotification(intl.formatMessage(messages.account_password_disabled)))
+  } catch (error) {
+    dispatch({ type: LOGIN_FAILURE, error: error.message })
+  }
+}
+
+/**
+ * requirePassword - Password protect routine. Should be placed before protected code.
+ *
+ * @param {string} password current password.
+ * @returns {Promise} Promise that fulfills after login attempt (either successful or not)
+ */
+const requirePassword = password => dispatch => {
+  return new Promise((resolve, reject) => {
+    dispatch(send('getPassword'))
```

Comment:
I'd suggest extracting this to a `getPassword` method which sends the message and listens/handles the relevant responses.

Reply:
wasn't addressed directly as per slack convo, but some refactoring was done